### PR TITLE
Use ROS_HOSTNAME before ROS_IP.

### PR DESCRIPTION
**Pull request by mikeodr**
To determine which interface to use follow ROS master convention
of `ROS_HOSTNAME` being used before `ROS_IP`.

This is as per:
http://wiki.ros.org/ROS/EnvironmentVariables#ROS_IP.2BAC8-ROS_HOSTNAME

### CODE REVIEW

**Comment by mikeodr** [comment]
@mikepurvis @paulbovbel FYI

File: master_discovery_fkie/src/master_discovery_fkie/udp.py [hunk 1]
```diff
@@ -189,10 +189,18 @@ def set_message_callback(self, callback):
...
+      # Otherwise use ROS_HOSTNAME before ROS_IP as per:
+      # http://wiki.ros.org/ROS/EnvironmentVariables#ROS_IP.2BAC8-ROS_HOSTNAME
+      if not interface and 'ROS_HOSTNAME' in os.environ:
+        addr = socket.gethostbyname(os.environ['ROS_HOSTNAME'])
+        # By default hostname resolves to localhost, don't set that
+        if addr != '127.0.0.1':
+          interface = addr
+      elif not interface and 'ROS_IP' in os.environ:
+          interface = os.environ['ROS_IP']
       return groupaddr, interface
```
Comment (mikepurvis) [inline master_discovery_fkie/src/master_discovery_fkie/udp.py]: I know opinions vary, but I have a slight preference in these scenarios for just doing early-return; IMO it makes the cascade more explicit:

```
if interface:
  return groupaddr, interface
if 'ROS_HOSTNAME' in os.environ:
  addr = socket.gethostbyname(os.environ['ROS_HOSTNAME'])
  if addr != '127.0.0.1':
    return groupaddr, addr
if 'ROS_IP' in os.environ:
  return groupaddr, os.environ['ROS_IP']
raise SomeError
```

**Comment by roehling** [comment]
I agree that early return makes the control flow a bit easier to grasp, so I took the liberty to change the code accordingly

**Comment by mikeodr** [comment]
Excellent thanks for the quick merge and fixup!

Is a bloom to 0.5.1 possible?

**Comment by roehling** [comment]
I'd prefer to leave the bloom release to @atiderko. He will be back in office next week.

**Comment by mikepurvis** [comment]
Much obliged, thanks to you both.

**Pull request merged**